crates/host/src/wasmbus/mod.rs

```diff
@@ -3944,6 +3961,18 @@ fn injector_to_headers(injector: &TraceContextInjector) -> async_nats::header::H
         .collect()
 }
 
+/// Helper function to ensure an individual instance's annotations contain all the
+/// filter annotations, determining if an instance matches a filter.
+///
+/// See #607 for more details.
+fn annotations_match_filter(annotations: &Annotations, filter: &Annotations) -> bool {
```

Comment:
nit; what if we had a function, which would take the instances and set of annotations to filter by and return an iterator instead?
This function could be as generic as desired, it could even be implemented as an extension on a generic actor instance collection, but all I'm trying to achieve here, is basically usage like this:
`actor_instances.filter_by_annotations(&annotations).for_each(...)`

it does not need to happen in this PR, but would be nice to perhaps add an issue to refactor this?

Reply:
I do like this idea, I couldn't quite get it to work nicely with my latest push in order to properly update the actors list as we remove across potentially different annotations. I think in general we really need to revisit this Annotations -> List of instances way of bookkeeping, once we can get rid of keeping track of instance counts this likely gets a whole lot easier.

For that reason I'd lean away from another issue, but this feedback should be considered when getting rid of the instances list.